download-directory/Jenkinsfile

```diff
@@ -0,0 +1,78 @@
+import groovy.json.JsonSlurper
+// This Jenkinsfile is used by Jenkins to run the DownloadDirectory step of Reactome's release.
+// It requires that the AddLinks-Insertion step has been run successfully before it can be run.
+pipeline {
+	agent any
+
+	stages {
+		// This stage checks that an upstream project, AddLinks-Insertion, was run successfully for its last build.
+		stage('Check AddLinks-Insertion build succeeded'){
+			steps{
+				script{
+					def currentRelease = (pwd() =~ /Releases\/(\d+)\//)[0][1];
```

Comment:
Does the value here differ from the environment variable `${env.RELEASE_NUMBER}`?

Reply:
This is actually what was causing me a headache one day a few weeks ago. I was using the environmental variable, but there was a blind spot created. If the user hasn't updated the release number to match the actual current release, what would end up happening is that it would end up checking the build status for the step of the _previous_ release instead of the current one. 

By doing it based on the current directory, which the user would have had to create themselves to run release, it removes the chance of a person falsely running a step based on the success of the step in the previous release.

Reply:
Ah okay, I see.  Perhaps the first step in the pipeline and/or a pre-step check common to all steps could compare the value obtained in the `currentRelease` variable, compare it to the environment variable, and throw an error if they are not the same?

That way the environment variable can always be used and it won't be a gotcha for any other places the environment variable is used.

Reply:
How it currently works is similar to that: Basically, each step has this check based on the working directory that should funnel you to the 'ConfirmReleaseConfigs' step, the now-first step of the release pipline when running it in Jenkins. ConfirmReleaseConfigs solicits your input for the release number, and then evaluates the environment's and the master configuration file's release number, confirming that all release number values are equal.

Reply:
Ah, okay I see.  Per our discussion, we could remove the release and previous release numbers from the environment variables and have them be determined through a function (and later on, possibly through a configuration local to the release run).